File: frontend/src/types/health.ts
export interface HealthResponse {
  status: string;
  app_name: string;
  version: string;
}

export interface ReadinessResponse {
  status: "ready" | "degraded";
  database: "up" | "down";
  redis: "up" | "down";
}

export interface VersionResponse {
  app_name: string;
  version: string;
  environment: string;
}

File: backend/app/schemas/health.py
"""Pydantic response models for health/readiness/version endpoints."""
from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: str
    app_name: str
    version: str


class ReadinessResponse(BaseModel):
    status: str
    database: str
    redis: str


class VersionResponse(BaseModel):
    app_name: str
    version: str
    environment: str
